traitlets/config/application.py

```diff
@@ -362,6 +367,17 @@ def print_help(self, classes=False):
 
         self.print_examples()
 
+    def document_config_options(self):
+        """Generate rST format documentation for the config options this application
+
+        Returns a multiline string.
+        """
+        chunks = []
+        for cls in self._classes_inc_parents():
+            chunks.append(cls.class_config_rst_doc())
+
+        return '\n'.join(chunks)
```

Comment:
I'm sad that `_classes_inc_parents` is a generator and that you for loop and not `map(cls, lambda cls:cls.class_config_rst_doc)`. I'm sure there is even a way to get rid of the lambda.